### DISCUSSION TITLE
two dimensional array with f64 elements can not store large number of entries

### DISCUSSION THREAD

**Discussion by flamehe0580** [vlang/v#17354] [Questions and Answers]
mut a := [][]f64{len: nrow, init: []f64{len: ncol, init: 0.0}}
here if nrow = 10000, and ncol = 100000, my computer system (win10 x64) pops up "Fatal error in GC Too many heap sections".
however, I believe this scale size of matrix is still not too large, because in other programming langeage, e.g. Julia, this size of arry will not be a problem in my computer.
So, i am asking for help how to fix this?

**Comment by flamehe0580** [comment]
here is the warning infomation if i used lager value of nrow and ncol:
Unhandled Exception 0x80000003
print_backtrace_skipping_top_frames is not implemented

**Reply by Wajinn** [reply]
Yeah, maybe something has changed, as had some pretty large 2d arrays before, and don't remember getting such an error. Definitely needs to be looked into.  This doesn't seem so much a discussion, as it should be a bug report.  Though developers might come by and offer some more insight.

Small example program to play with.

```
fn main() {
	nrow := 10000
	ncol := 100000 
	
	arr_2d := [][]f64{len: nrow, init: []f64{len: ncol}}
 
	println(arr_2d.len)
}
```

**Reply by flamehe0580** [reply]
fn main() {
	nrow := 10000
	ncol := 100000 
	
	arr_2d := [][]f64{len: nrow, init: []f64{len: ncol}}
 
	println(arr_2d.len)
}
this code will also triger "fatal error in GC: Too many heap sections " in my computer.

**Comment by JalonSolov** [comment]
Which OS?

On Manjaro Linux...
```
$ v run foo.v
10000
$
```

**Reply by flamehe0580** [reply]
windows 10 x64

**Reply by JalonSolov** [reply]
It's quite possible you're simply blowing out the stack.  Windows has a horribly small default stack compared to other OSes.

Try compiling with `v -thread-stack-size=16777216 yourfile.v`

**Reply by flamehe0580** [reply]
it seems that the dynamic array here i used is allocated in the heap memory, and the fixed size array [1000][100000]f64{} may be controlled by  parameter "-thread-stack-size 167772160".
And i tried "[1000][100000]f64{}" and using "v -thread-stack-size 167772160 -prod main.v", which resulted in correct results. However, it will not succeed if the "-prod" is neglected.
now, i am totally confused the correct usage of vlang.

**Reply by Wajinn** [reply]
> And i tried "[1000][100000]f64{}" and using "`v -thread-stack-size 167772160 -prod main.v`", which resulted in correct results. 

Well, people are giving a workaround solution that can be applied at this moment.  You tested it, and confirmed that it can work.

> However, it will not succeed if the "-prod" is neglected. now, i am totally confused the correct usage of vlang.

I'm confused, by the confusion.  JalonSolov stated that he thinks the issue might be that Windows has a small default stack compared to other OSes.  Looks like 2d arrays above a certain size weren't tested for the Windows OS, so now much larger multidimensional array tests are needed.  If that's the case, and it's an issue affecting specifically the Windows OS, then Vlang developers should be able to address it in newer versions of V.    It's actually a good thing, that you discovered the issue, and brought attention to it.  

Perhaps you want to file a bug/feature request, as it clearly is an issue.

**Reply by flamehe0580** [reply]
this problem is finally solved (in windows 10 x86_64) by "v -cc msvc file_name.v", using msvc as the compiler. 
The compiled binary file by "v -cc msvc file_name.v " can run normally.
However, "-cc gcc" will fail due to the large size of array.

**Reply by Wajinn** [reply]
> this problem is finally solved (in windows 10 x86_64) by "v -cc msvc file_name.v", using msvc as the compiler. The compiled binary file by "v -cc msvc file_name.v " can run normally.

Still, kind of think this should be turned into a bug or feature request, because it might be possible for V developers to solve it on the Winows OS using the default tcc.

**Reply by spytheman** [reply]
> Still, kind of think this should be turned into a bug or feature request, because it might be possible for V developers to solve it on the Winows OS using the default tcc.

Done in https://github.com/vlang/v/issues/17736 .
